D/D.sublime-syntax

```diff
@@ -8,339 +8,2057 @@ file_extensions:
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.d
+        2: constant.numeric.integer.hexadecimal.d
+        3: storage.type.integer.d
+      pop: true
+
+  floating-point:
+    - include: floating-point-opt
+    - include: not-whitespace-illegal
+  floating-point-opt:
+    - match: '\b(0[xX])({{hex_or_under}}*\.?{{hex_or_under}}+)([pP][+\-]?[0-9]+)({{float_suffix}})?\b'
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.d
+        2: constant.numeric.float.hexadecimal.d
+        3: constant.numeric.float.hexadecimal.d
+        4: storage.type.float.d
+      pop: true
+    - match: '\b(0|[1-9][0-9_]*)({{integer_float_suffix}})\b'
+      captures:
+        1: constant.numeric.float.d
+        2: storage.type.float.d
+      pop: true
+    - match: '\b(0[bB])([01][01_]*)({{integer_float_suffix}})\b'
+      captures:
+        1: punctuation.definition.numeric.binary.d
+        2: constant.numeric.float.binary.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b(0[xX])({{hex_char}}{{hex_or_under}}*)({{integer_float_suffix}})\b'
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.d
+        2: constant.numeric.float.hexadecimal.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b([0-9][0-9_]*\.[0-9][0-9_]*)({{float_exponent}})?({{float_suffix}})?\b'
+      captures:
+        1: constant.numeric.float.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b([0-9][0-9_]*\.(?!\.|\s*[a-zA-Z]))({{float_suffix}})?'
       captures:
-        1: storage.modifier.d
-        2: storage.type.d
-        3: entity.name.function.d
+        1: constant.numeric.float.d
+        2: storage.type.float.d
+      pop: true
+    - match: '(\.[0-9][0-9_]*)({{float_exponent}})?({{float_suffix}})?\b'
+      captures:
+        1: constant.numeric.float.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b([0-9][0-9_]*)({{float_exponent}})({{float_suffix}})?\b'
+      captures:
+        1: constant.numeric.float.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+
+  character-in:
+    - match: '{{character_lookahead}}'
+      push: character
+  character:
+    - match: \'(?:({{escape_sequence}})|[^\\'])\'
+      scope: string.quoted.single.d
+      captures:
+        1: constant.character.escape.d
+      pop: true
+    - match: \'[^']*\'
+      scope: invalid.illegal.d
+      pop: true
+    - include: not-whitespace-illegal-pop
+
+  string-in:
+    - match: '{{string_lookahead}}'
+      push: string
+  string:
+    - include: string-opt
+    - include: not-whitespace-illegal
+  string-opt:
+    # Regular string
+    - match: '"'
+      scope: punctuation.definition.string.begin.d
+      set:
+        - meta_include_prototype: false
+        - meta_scope: string.quoted.double.d
```

Comment:
In addition to `string.quoted`, each string should have `meta.string` that starts at the very first token that begins the string and continues until the very last. This is useful when dealing with interpolations since the `string.quoted` scope will normally be popped off the stack to allow the interpolated expression to show through without all of the colors from a string.